File: AIRLINE_RESERVATION/TakeOff/src/main/java/com/air/appl/services/BookingService.java
package com.air.appl.services;

import java.util.Date;
import java.util.List;

import com.air.appl.beans.Booking;
import com.air.appl.beans.Flight;
import com.air.appl.beans.User;
import com.air.appl.beans.sFlight;

public interface BookingService {

//	public List<Flight> searchFlight(String source, String destination,Date departureDate,String travelClass);
	public List<Flight> searchFlight(sFlight sf);
	public Booking addBooking(Booking b);
	public List<Booking> viewBooking(int userId);
	//public long addBookings(Booking b,String travelClass, int flightId, int userId);
	public String deleteBooking(int bookingId);

}

File: AIRLINE_RESERVATION/User_final/src/app/Service/booking.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Payment } from '../user/Payment';
import { sFlight } from '../user/sFlight';

@Injectable({
  providedIn: 'root'
})
export class BookingService {
  baseUrl: string="http://localhost:8090/api/v1/";


  constructor(private http:HttpClient) { }

  public searchFlight1(sflight: Object) :Observable<Object> {
    
    // console.log(( "http://localhost:8090/api/v1/" +source+"/"+destination+"/"+departureDate+"/"+travelClass));
   return this.http.post( "http://localhost:8090/api/v1/sFlight" , sflight); }
 


   public addBooking(booking:object) :Observable<Object>
   {
     return this.http.post("http://localhost:8090/api/v1/addBooking", booking);
   }

   public addPassenger(passenger:object) :Observable<Object>
   {
     return this.http.post("http://localhost:8090/api/v1/addPassenger", passenger)
   }

   public payBill(payment:object) :Observable<Object>
   {
      return this.http.post("http://localhost:8090/api/v1/payBill", payment)
   }

   public viewBooking (userId:number)
   {
     return this.http.get("http://localhost:8090/api/v1/bookings/" + userId);
   }

   public cancelBooking(bookingId:number)
   {
     return this.http.get("http://localhost:8090/api/v1/bookingdelete/" + bookingId);
   }
}
